In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('fodder1.csv')
df.head()

SchemeCode                                         SchemeName  \
0    100033.0  Aditya Birla Sun Life Equity Advantage Fund Gr...   
1    100034.0  Aditya Birla Sun Life Equity Advantage Fund Pa...   
2    100037.0  Aditya Birla Sun Life Income Fund Regular Plan...   
3    100038.0  Aditya Birla Sun Life Income Fund Regular Plan...   
4    100077.0  DSP Bond Fund Regular Plan Reinvestment of Inc...   

            SchemeCategory   Inception               ExpenseRatio  \
0         Large & Mid- Cap  24/02/1995  1.9500 (as on 28/02/2025)   
1         Large & Mid- Cap  24/02/1995  1.9500 (as on 28/02/2025)   
2  Medium to Long Duration  23/10/1995  1.1600 (as on 28/02/2025)   
3  Medium to Long Duration  21/10/1995  1.1600 (as on 28/02/2025)   
4          Medium Duration  29/04/1997  0.7500 (as on 28/02/2025)   

                TurnoverRatio  ALP3Y  ALP5Y ALP10Y BET3Y  ... BET10Y R2_3YRS  \
0   40.0000(as on 28/02/2025)   -4.5  -2.44  -2.13  1.04  ...   1.07   88.31   
1   40.0000(as on 28/02/2025)  -4.95  -2.76   -2.5  1.04  ...   1.07   88.71   
2  633.9800(as on 31/03/2024)    NaN    NaN    NaN   NaN  ...    NaN     NaN   
3  633.9800(as on 31/03/2024)    NaN    NaN    NaN   NaN  ...    NaN     NaN   
4  170.9400(as on 31/03/2024)    NaN    NaN    NaN   NaN  ...    NaN     NaN   

  R2_5YRS R2_10YRS SD_3YRS SD_5YRS SD_10YRS SHP_3YRS SHP_5YRS SHP_10YRS  
0   88.06    90.02   16.12   16.66    18.82      0.2     0.99      0.34  
1   88.21    89.96   16.09   16.65    18.78     0.17     0.97      0.32  
2     NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  
3     NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  
4     NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  

[5 rows x 21 columns]

In [3]:
df.describe()

SchemeCode
count    1170.000000
mean   111684.539316
std      7556.823280
min    100033.000000
25%    103192.250000
50%    112877.500000
75%    119016.750000
max    120482.000000

In [4]:
df

SchemeCode                                         SchemeName  \
0       100033.0  Aditya Birla Sun Life Equity Advantage Fund Gr...   
1       100034.0  Aditya Birla Sun Life Equity Advantage Fund Pa...   
2       100037.0  Aditya Birla Sun Life Income Fund Regular Plan...   
3       100038.0  Aditya Birla Sun Life Income Fund Regular Plan...   
4       100077.0  DSP Bond Fund Regular Plan Reinvestment of Inc...   
...          ...                                                ...   
1169    120478.0  Axis Conservative Hybrid Fund Direct Plan Half...   
1170    120479.0  Axis Conservative Hybrid Fund Direct Plan Quar...   
1171    120480.0  Axis Conservative Hybrid Fund Direct Plan Grow...   
1172    120481.0  JM Arbitrage Fund (Direct) Reinvestment of Inc...   
1173    120482.0         JM Arbitrage Fund (Direct) - Growth Option   

               SchemeCategory   Inception               ExpenseRatio  \
0            Large & Mid- Cap  24/02/1995  1.9500 (as on 28/02/2025)   
1            Large & Mid- Cap  24/02/1995  1.9500 (as on 28/02/2025)   
2     Medium to Long Duration  23/10/1995  1.1600 (as on 28/02/2025)   
3     Medium to Long Duration  21/10/1995  1.1600 (as on 28/02/2025)   
4             Medium Duration  29/04/1997  0.7500 (as on 28/02/2025)   
...                       ...         ...                        ...   
1169  Conservative Allocation  05/07/2013  0.9000 (as on 31/03/2025)   
1170  Conservative Allocation  21/01/2013  0.9000 (as on 31/03/2025)   
1171  Conservative Allocation  01/01/2013  0.9000 (as on 31/03/2025)   
1172           Arbitrage Fund  01/01/2013  0.4000 (as on 31/03/2025)   
1173           Arbitrage Fund  01/01/2013  0.4000 (as on 31/03/2025)   

                    TurnoverRatio  ALP3Y  ALP5Y ALP10Y BET3Y  ... BET10Y  \
0       40.0000(as on 28/02/2025)   -4.5  -2.44  -2.13  1.04  ...   1.07   
1       40.0000(as on 28/02/2025)  -4.95  -2.76   -2.5  1.04  ...   1.07   
2      633.9800(as on 31/03/2024)    NaN    NaN    NaN   NaN  ...    NaN   
3      633.9800(as on 31/03/2024)    NaN    NaN    NaN   NaN  ...    NaN   
4      170.9400(as on 31/03/2024)    NaN    NaN    NaN   NaN  ...    NaN   
...                           ...    ...    ...    ...   ...  ...    ...   
1169   126.9100(as on 31/03/2024)  -0.86   1.23  -1.26  1.18  ...   0.98   
1170   126.9100(as on 31/03/2024)  -0.85   1.19  -1.19  1.17  ...   1.01   
1171   126.9100(as on 31/03/2024)  -0.69   0.68  -0.39  1.18  ...   0.98   
1172  1123.9500(as on 31/03/2025)    NaN    NaN    NaN   NaN  ...    NaN   
1173  1123.9500(as on 31/03/2025)    NaN    NaN    NaN   NaN  ...    NaN   

     R2_3YRS R2_5YRS R2_10YRS SD_3YRS SD_5YRS SD_10YRS SHP_3YRS SHP_5YRS  \
0      88.31   88.06    90.02   16.12   16.66    18.82      0.2     0.99   
1      88.71   88.21    89.96   16.09   16.65    18.78     0.17     0.97   
2        NaN     NaN      NaN     NaN     NaN      NaN      NaN      NaN   
3        NaN     NaN      NaN     NaN     NaN      NaN      NaN      NaN   
4        NaN     NaN      NaN     NaN     NaN      NaN      NaN      NaN   
...      ...     ...      ...     ...     ...      ...      ...      ...   
1169   89.23   77.74    50.62    4.32    4.47     4.98     0.43     1.20   
1170   89.03   78.16    53.31    4.30    4.43     4.98     0.43     1.20   
1171   88.91   82.19    57.04    4.32    4.27     4.71     0.46     1.11   
1172     NaN     NaN      NaN     NaN     NaN      NaN      NaN      NaN   
1173     NaN     NaN      NaN     NaN     NaN      NaN      NaN      NaN   

     SHP_10YRS  
0         0.34  
1         0.32  
2          NaN  
3          NaN  
4          NaN  
...        ...  
1169      0.29  
1170      0.32  
1171      0.49  
1172       NaN  
1173       NaN  

[1174 rows x 21 columns]

In [5]:
df.columns

Index(['SchemeCode', 'SchemeName', 'SchemeCategory', 'Inception',
       'ExpenseRatio', 'TurnoverRatio', 'ALP3Y', 'ALP5Y', 'ALP10Y', 'BET3Y',
       'BET5Y', 'BET10Y', 'R2_3YRS', 'R2_5YRS', 'R2_10YRS', 'SD_3YRS',
       'SD_5YRS', 'SD_10YRS', 'SHP_3YRS', 'SHP_5YRS', 'SHP_10YRS'],
      dtype='object')

In [6]:
df.SchemeCategory.value_counts()

Low Duration                130
Conservative Allocation      96
Short Duration               83
Large-Cap                    82
Money Market                 70
Large & Mid- Cap             68
Index Funds                  61
Medium to Long Duration      59
Mid-Cap                      57
Ultra Short Duration         54
Aggressive Allocation        53
Medium Duration              49
Corporate Bond               48
Banking & PSU                43
Arbitrage Fund               41
Value                        34
Small-Cap                    30
Multi Asset Allocation       22
Multi-Cap                    18
Dynamic Asset Allocation     16
Contra                       12
10 yr Government Bond        12
Equity Savings               12
Other Bond                    9
Long Duration                 2
Balanced Allocation           1
Name: SchemeCategory, dtype: int64

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1174 entries, 0 to 1173
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   SchemeCode      1170 non-null   float64
 1   SchemeName      1162 non-null   object 
 2   SchemeCategory  1162 non-null   object 
 3   Inception       1162 non-null   object 
 4   ExpenseRatio    1162 non-null   object 
 5   TurnoverRatio   1162 non-null   object 
 6   ALP3Y           911 non-null    object 
 7   ALP5Y           911 non-null    object 
 8   ALP10Y          911 non-null    object 
 9   BET3Y           911 non-null    object 
 10  BET5Y           911 non-null    object 
 11  BET10Y          911 non-null    object 
 12  R2_3YRS         911 non-null    object 
 13  R2_5YRS         911 non-null    object 
 14  R2_10YRS        911 non-null    object 
 15  SD_3YRS         911 non-null    object 
 16  SD_5YRS         911 non-null    object 
 17  SD_10YRS        911 non-null    o

## Data Cleaning

In [8]:
#check for null values
df.isnull().sum()

SchemeCode          4
SchemeName         12
SchemeCategory     12
Inception          12
ExpenseRatio       12
TurnoverRatio      12
ALP3Y             263
ALP5Y             263
ALP10Y            263
BET3Y             263
BET5Y             263
BET10Y            263
R2_3YRS           263
R2_5YRS           263
R2_10YRS          263
SD_3YRS           263
SD_5YRS           263
SD_10YRS          263
SHP_3YRS          263
SHP_5YRS          263
SHP_10YRS         263
dtype: int64

In [9]:
df[df['SchemeCategory'] == 'Index Funds']

SchemeCode                                         SchemeName  \
53      100484.0      Franklin India NSE Nifty 50 Index Fund Growth   
98      100822.0        UTI Nifty 50 Index Fund Regular Plan Growth   
99      100823.0  UTI Nifty 50 Index Fund Regular Plan Payout In...   
126     101198.0  LIC MF BSE Sensex Index Payout of Income Dis c...   
127     101199.0                     LIC MF BSE Sensex Index Growth   
...          ...                                                ...   
1058    119827.0                 SBI Nifty Index Fund Direct Growth   
1114    120307.0  LIC MF Nifty 50 Index Fund Direct Plan Growth ...   
1115    120308.0  LIC MF BSE Sensex Index Direct Plan Growth option   
1116    120309.0  LIC MF Nifty 50 Index Fund Direct Plan Reinves...   
1117    120312.0  LIC MF BSE Sensex Index Direct Plan Reinvestme...   

     SchemeCategory   Inception               ExpenseRatio  \
53      Index Funds  04/08/2000  0.6300 (as on 28/02/2025)   
98      Index Funds  06/03/2000  0.2500 (as on 28/02/2025)   
99      Index Funds  06/03/2000  0.2500 (as on 28/02/2025)   
126     Index Funds  14/11/2002  1.0800 (as on 28/02/2025)   
127     Index Funds  14/11/2002  1.0800 (as on 28/02/2025)   
...             ...         ...                        ...   
1058    Index Funds  01/01/2013  0.2200 (as on 31/03/2025)   
1114    Index Funds  01/01/2013  0.1800 (as on 31/03/2025)   
1115    Index Funds  01/01/2013  0.3800 (as on 31/03/2025)   
1116    Index Funds  01/01/2013  0.1800 (as on 31/03/2025)   
1117    Index Funds  01/01/2013  0.3800 (as on 31/03/2025)   

                  TurnoverRatio ALP3Y ALP5Y ALP10Y BET3Y  ... BET10Y R2_3YRS  \
53     8.3400(as on 31/03/2024)   NaN   NaN    NaN   NaN  ...    NaN     NaN   
98     6.0000(as on 28/02/2025)   NaN   NaN    NaN   NaN  ...    NaN     NaN   
99     6.0000(as on 28/02/2025)   NaN   NaN    NaN   NaN  ...    NaN     NaN   
126   14.0000(as on 28/02/2025)   NaN   NaN    NaN   NaN  ...    NaN     NaN   
127   14.0000(as on 28/02/2025)   NaN   NaN    NaN   NaN  ...    NaN     NaN   
...                         ...   ...   ...    ...   ...  ...    ...     ...   
1058  13.0000(as on 31/03/2025)   NaN   NaN    NaN   NaN  ...    NaN     NaN   
1114   9.0000(as on 31/03/2025)   NaN   NaN    NaN   NaN  ...    NaN     NaN   
1115  12.0000(as on 31/03/2025)   NaN   NaN    NaN   NaN  ...    NaN     NaN   
1116   9.0000(as on 31/03/2025)   NaN   NaN    NaN   NaN  ...    NaN     NaN   
1117  12.0000(as on 31/03/2025)   NaN   NaN    NaN   NaN  ...    NaN     NaN   

     R2_5YRS R2_10YRS SD_3YRS SD_5YRS SD_10YRS SHP_3YRS SHP_5YRS SHP_10YRS  
53       NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  
98       NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  
99       NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  
126      NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  
127      NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  
...      ...      ...     ...     ...      ...      ...      ...       ...  
1058     NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  
1114     NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  
1115     NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  
1116     NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  
1117     NaN      NaN     NaN     NaN      NaN      NaN      NaN       NaN  

[61 rows x 21 columns]

In [10]:
#drop missing values
df_clean = df.dropna()
df_clean

SchemeCode                                         SchemeName  \
0       100033.0  Aditya Birla Sun Life Equity Advantage Fund Gr...   
1       100034.0  Aditya Birla Sun Life Equity Advantage Fund Pa...   
12      100176.0  Quant Small Cap Fund Payout of Income Distribu...   
13      100177.0                        Quant Small Cap Fund Growth   
14      100218.0  JM Large Cap Fund Payout of Income Distributio...   
...          ...                                                ...   
1167    120475.0      Axis Strategic Bond Fund Direct Growth Option   
1168    120477.0  Axis Conservative Hybrid Fund Direct Plan Annu...   
1169    120478.0  Axis Conservative Hybrid Fund Direct Plan Half...   
1170    120479.0  Axis Conservative Hybrid Fund Direct Plan Quar...   
1171    120480.0  Axis Conservative Hybrid Fund Direct Plan Grow...   

               SchemeCategory   Inception               ExpenseRatio  \
0            Large & Mid- Cap  24/02/1995  1.9500 (as on 28/02/2025)   
1            Large & Mid- Cap  24/02/1995  1.9500 (as on 28/02/2025)   
12                  Small-Cap  31/10/1996  1.6400 (as on 31/03/2025)   
13                  Small-Cap  23/09/1996  1.6400 (as on 31/03/2025)   
14                  Large-Cap  01/04/1995  2.2500 (as on 28/02/2025)   
...                       ...         ...                        ...   
1167          Medium Duration  01/01/2013  0.5300 (as on 31/03/2025)   
1168  Conservative Allocation  11/09/2013  0.9000 (as on 31/03/2025)   
1169  Conservative Allocation  05/07/2013  0.9000 (as on 31/03/2025)   
1170  Conservative Allocation  21/01/2013  0.9000 (as on 31/03/2025)   
1171  Conservative Allocation  01/01/2013  0.9000 (as on 31/03/2025)   

                   TurnoverRatio  ALP3Y  ALP5Y ALP10Y BET3Y  ... BET10Y  \
0      40.0000(as on 28/02/2025)   -4.5  -2.44  -2.13  1.04  ...   1.07   
1      40.0000(as on 28/02/2025)  -4.95  -2.76   -2.5  1.04  ...   1.07   
12    151.0000(as on 30/11/2023)   2.44  10.06   4.59  0.88  ...   0.71   
13    151.0000(as on 30/11/2023)    2.4  10.01   4.69  0.88  ...   0.72   
14    276.1600(as on 28/02/2025)   0.62  -0.98  -1.61  0.96  ...    0.7   
...                          ...    ...    ...    ...   ...  ...    ...   
1167  269.2700(as on 31/03/2024)   1.10   1.34   0.54  1.19  ...   1.26   
1168  126.9100(as on 31/03/2024)  -0.80   0.59  -1.38  1.17  ...   0.92   
1169  126.9100(as on 31/03/2024)  -0.86   1.23  -1.26  1.18  ...   0.98   
1170  126.9100(as on 31/03/2024)  -0.85   1.19  -1.19  1.17  ...   1.01   
1171  126.9100(as on 31/03/2024)  -0.69   0.68  -0.39  1.18  ...   0.98   

     R2_3YRS R2_5YRS R2_10YRS SD_3YRS SD_5YRS SD_10YRS SHP_3YRS SHP_5YRS  \
0      88.31   88.06    90.02   16.12   16.66    18.82      0.2     0.99   
1      88.71   88.21    89.96   16.09   16.65    18.78     0.17     0.97   
12     89.49   73.71    59.24   18.84   23.07    20.99     0.75     1.66   
13     89.51   73.74    59.45   18.84   23.07    20.98     0.75     1.66   
14     88.22   82.52    85.55    14.3   13.04    12.47      0.5      1.0   
...      ...     ...      ...     ...     ...      ...      ...      ...   
1167   83.91   83.11    62.70    1.37    1.55     2.19     1.27     1.67   
1168   88.87   82.12    47.05    4.30    4.26     4.85     0.44     1.09   
1169   89.23   77.74    50.62    4.32    4.47     4.98     0.43     1.20   
1170   89.03   78.16    53.31    4.30    4.43     4.98     0.43     1.20   
1171   88.91   82.19    57.04    4.32    4.27     4.71     0.46     1.11   

     SHP_10YRS  
0         0.34  
1         0.32  
12        0.64  
13        0.64  
14        0.29  
...        ...  
1167      1.05  
1168      0.24  
1169      0.29  
1170      0.32  
1171      0.49  

[911 rows x 21 columns]

In [11]:
df_clean = df_clean.reset_index(drop=True)
df_clean

SchemeCode                                         SchemeName  \
0      100033.0  Aditya Birla Sun Life Equity Advantage Fund Gr...   
1      100034.0  Aditya Birla Sun Life Equity Advantage Fund Pa...   
2      100176.0  Quant Small Cap Fund Payout of Income Distribu...   
3      100177.0                        Quant Small Cap Fund Growth   
4      100218.0  JM Large Cap Fund Payout of Income Distributio...   
..          ...                                                ...   
906    120475.0      Axis Strategic Bond Fund Direct Growth Option   
907    120477.0  Axis Conservative Hybrid Fund Direct Plan Annu...   
908    120478.0  Axis Conservative Hybrid Fund Direct Plan Half...   
909    120479.0  Axis Conservative Hybrid Fund Direct Plan Quar...   
910    120480.0  Axis Conservative Hybrid Fund Direct Plan Grow...   

              SchemeCategory   Inception               ExpenseRatio  \
0           Large & Mid- Cap  24/02/1995  1.9500 (as on 28/02/2025)   
1           Large & Mid- Cap  24/02/1995  1.9500 (as on 28/02/2025)   
2                  Small-Cap  31/10/1996  1.6400 (as on 31/03/2025)   
3                  Small-Cap  23/09/1996  1.6400 (as on 31/03/2025)   
4                  Large-Cap  01/04/1995  2.2500 (as on 28/02/2025)   
..                       ...         ...                        ...   
906          Medium Duration  01/01/2013  0.5300 (as on 31/03/2025)   
907  Conservative Allocation  11/09/2013  0.9000 (as on 31/03/2025)   
908  Conservative Allocation  05/07/2013  0.9000 (as on 31/03/2025)   
909  Conservative Allocation  21/01/2013  0.9000 (as on 31/03/2025)   
910  Conservative Allocation  01/01/2013  0.9000 (as on 31/03/2025)   

                  TurnoverRatio  ALP3Y  ALP5Y ALP10Y BET3Y  ... BET10Y  \
0     40.0000(as on 28/02/2025)   -4.5  -2.44  -2.13  1.04  ...   1.07   
1     40.0000(as on 28/02/2025)  -4.95  -2.76   -2.5  1.04  ...   1.07   
2    151.0000(as on 30/11/2023)   2.44  10.06   4.59  0.88  ...   0.71   
3    151.0000(as on 30/11/2023)    2.4  10.01   4.69  0.88  ...   0.72   
4    276.1600(as on 28/02/2025)   0.62  -0.98  -1.61  0.96  ...    0.7   
..                          ...    ...    ...    ...   ...  ...    ...   
906  269.2700(as on 31/03/2024)   1.10   1.34   0.54  1.19  ...   1.26   
907  126.9100(as on 31/03/2024)  -0.80   0.59  -1.38  1.17  ...   0.92   
908  126.9100(as on 31/03/2024)  -0.86   1.23  -1.26  1.18  ...   0.98   
909  126.9100(as on 31/03/2024)  -0.85   1.19  -1.19  1.17  ...   1.01   
910  126.9100(as on 31/03/2024)  -0.69   0.68  -0.39  1.18  ...   0.98   

    R2_3YRS R2_5YRS R2_10YRS SD_3YRS SD_5YRS SD_10YRS SHP_3YRS SHP_5YRS  \
0     88.31   88.06    90.02   16.12   16.66    18.82      0.2     0.99   
1     88.71   88.21    89.96   16.09   16.65    18.78     0.17     0.97   
2     89.49   73.71    59.24   18.84   23.07    20.99     0.75     1.66   
3     89.51   73.74    59.45   18.84   23.07    20.98     0.75     1.66   
4     88.22   82.52    85.55    14.3   13.04    12.47      0.5      1.0   
..      ...     ...      ...     ...     ...      ...      ...      ...   
906   83.91   83.11    62.70    1.37    1.55     2.19     1.27     1.67   
907   88.87   82.12    47.05    4.30    4.26     4.85     0.44     1.09   
908   89.23   77.74    50.62    4.32    4.47     4.98     0.43     1.20   
909   89.03   78.16    53.31    4.30    4.43     4.98     0.43     1.20   
910   88.91   82.19    57.04    4.32    4.27     4.71     0.46     1.11   

    SHP_10YRS  
0        0.34  
1        0.32  
2        0.64  
3        0.64  
4        0.29  
..        ...  
906      1.05  
907      0.24  
908      0.29  
909      0.32  
910      0.49  

[911 rows x 21 columns]

In [12]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 911 entries, 0 to 910
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   SchemeCode      911 non-null    float64
 1   SchemeName      911 non-null    object 
 2   SchemeCategory  911 non-null    object 
 3   Inception       911 non-null    object 
 4   ExpenseRatio    911 non-null    object 
 5   TurnoverRatio   911 non-null    object 
 6   ALP3Y           911 non-null    object 
 7   ALP5Y           911 non-null    object 
 8   ALP10Y          911 non-null    object 
 9   BET3Y           911 non-null    object 
 10  BET5Y           911 non-null    object 
 11  BET10Y          911 non-null    object 
 12  R2_3YRS         911 non-null    object 
 13  R2_5YRS         911 non-null    object 
 14  R2_10YRS        911 non-null    object 
 15  SD_3YRS         911 non-null    object 
 16  SD_5YRS         911 non-null    object 
 17  SD_10YRS        911 non-null    obj

In [13]:
df_clean.SchemeCategory.value_counts()

Low Duration               130
Conservative Allocation     88
Large-Cap                   82
Short Duration              77
Large & Mid- Cap            68
Money Market                64
Mid-Cap                     57
Ultra Short Duration        51
Banking & PSU               41
Medium Duration             39
Medium to Long Duration     39
Aggressive Allocation       36
Value                       34
Corporate Bond              34
Small-Cap                   30
Multi-Cap                   17
10 yr Government Bond       12
Contra                      12
Name: SchemeCategory, dtype: int64

In [14]:
df_clean

SchemeCode                                         SchemeName  \
0      100033.0  Aditya Birla Sun Life Equity Advantage Fund Gr...   
1      100034.0  Aditya Birla Sun Life Equity Advantage Fund Pa...   
2      100176.0  Quant Small Cap Fund Payout of Income Distribu...   
3      100177.0                        Quant Small Cap Fund Growth   
4      100218.0  JM Large Cap Fund Payout of Income Distributio...   
..          ...                                                ...   
906    120475.0      Axis Strategic Bond Fund Direct Growth Option   
907    120477.0  Axis Conservative Hybrid Fund Direct Plan Annu...   
908    120478.0  Axis Conservative Hybrid Fund Direct Plan Half...   
909    120479.0  Axis Conservative Hybrid Fund Direct Plan Quar...   
910    120480.0  Axis Conservative Hybrid Fund Direct Plan Grow...   

              SchemeCategory   Inception               ExpenseRatio  \
0           Large & Mid- Cap  24/02/1995  1.9500 (as on 28/02/2025)   
1           Large & Mid- Cap  24/02/1995  1.9500 (as on 28/02/2025)   
2                  Small-Cap  31/10/1996  1.6400 (as on 31/03/2025)   
3                  Small-Cap  23/09/1996  1.6400 (as on 31/03/2025)   
4                  Large-Cap  01/04/1995  2.2500 (as on 28/02/2025)   
..                       ...         ...                        ...   
906          Medium Duration  01/01/2013  0.5300 (as on 31/03/2025)   
907  Conservative Allocation  11/09/2013  0.9000 (as on 31/03/2025)   
908  Conservative Allocation  05/07/2013  0.9000 (as on 31/03/2025)   
909  Conservative Allocation  21/01/2013  0.9000 (as on 31/03/2025)   
910  Conservative Allocation  01/01/2013  0.9000 (as on 31/03/2025)   

                  TurnoverRatio  ALP3Y  ALP5Y ALP10Y BET3Y  ... BET10Y  \
0     40.0000(as on 28/02/2025)   -4.5  -2.44  -2.13  1.04  ...   1.07   
1     40.0000(as on 28/02/2025)  -4.95  -2.76   -2.5  1.04  ...   1.07   
2    151.0000(as on 30/11/2023)   2.44  10.06   4.59  0.88  ...   0.71   
3    151.0000(as on 30/11/2023)    2.4  10.01   4.69  0.88  ...   0.72   
4    276.1600(as on 28/02/2025)   0.62  -0.98  -1.61  0.96  ...    0.7   
..                          ...    ...    ...    ...   ...  ...    ...   
906  269.2700(as on 31/03/2024)   1.10   1.34   0.54  1.19  ...   1.26   
907  126.9100(as on 31/03/2024)  -0.80   0.59  -1.38  1.17  ...   0.92   
908  126.9100(as on 31/03/2024)  -0.86   1.23  -1.26  1.18  ...   0.98   
909  126.9100(as on 31/03/2024)  -0.85   1.19  -1.19  1.17  ...   1.01   
910  126.9100(as on 31/03/2024)  -0.69   0.68  -0.39  1.18  ...   0.98   

    R2_3YRS R2_5YRS R2_10YRS SD_3YRS SD_5YRS SD_10YRS SHP_3YRS SHP_5YRS  \
0     88.31   88.06    90.02   16.12   16.66    18.82      0.2     0.99   
1     88.71   88.21    89.96   16.09   16.65    18.78     0.17     0.97   
2     89.49   73.71    59.24   18.84   23.07    20.99     0.75     1.66   
3     89.51   73.74    59.45   18.84   23.07    20.98     0.75     1.66   
4     88.22   82.52    85.55    14.3   13.04    12.47      0.5      1.0   
..      ...     ...      ...     ...     ...      ...      ...      ...   
906   83.91   83.11    62.70    1.37    1.55     2.19     1.27     1.67   
907   88.87   82.12    47.05    4.30    4.26     4.85     0.44     1.09   
908   89.23   77.74    50.62    4.32    4.47     4.98     0.43     1.20   
909   89.03   78.16    53.31    4.30    4.43     4.98     0.43     1.20   
910   88.91   82.19    57.04    4.32    4.27     4.71     0.46     1.11   

    SHP_10YRS  
0        0.34  
1        0.32  
2        0.64  
3        0.64  
4        0.29  
..        ...  
906      1.05  
907      0.24  
908      0.29  
909      0.32  
910      0.49  

[911 rows x 21 columns]

## Feature Engineering

In [15]:
df_clean['Inception'] = pd.to_datetime(df_clean['Inception'])

C:\Users\Chaitanya Keshari\AppData\Local\Temp\ipykernel_22964\1757003806.py:1: UserWarning: Parsing dates in DD/MM/YYYY format when dayfirst=False (the default) was specified. This may lead to inconsistently parsed dates! Specify a format to ensure consistent parsing.
  df_clean['Inception'] = pd.to_datetime(df_clean['Inception'])


In [16]:
df_clean.head()

SchemeCode                                         SchemeName  \
0    100033.0  Aditya Birla Sun Life Equity Advantage Fund Gr...   
1    100034.0  Aditya Birla Sun Life Equity Advantage Fund Pa...   
2    100176.0  Quant Small Cap Fund Payout of Income Distribu...   
3    100177.0                        Quant Small Cap Fund Growth   
4    100218.0  JM Large Cap Fund Payout of Income Distributio...   

     SchemeCategory  Inception               ExpenseRatio  \
0  Large & Mid- Cap 1995-02-24  1.9500 (as on 28/02/2025)   
1  Large & Mid- Cap 1995-02-24  1.9500 (as on 28/02/2025)   
2         Small-Cap 1996-10-31  1.6400 (as on 31/03/2025)   
3         Small-Cap 1996-09-23  1.6400 (as on 31/03/2025)   
4         Large-Cap 1995-01-04  2.2500 (as on 28/02/2025)   

                TurnoverRatio  ALP3Y  ALP5Y ALP10Y BET3Y  ... BET10Y R2_3YRS  \
0   40.0000(as on 28/02/2025)   -4.5  -2.44  -2.13  1.04  ...   1.07   88.31   
1   40.0000(as on 28/02/2025)  -4.95  -2.76   -2.5  1.04  ...   1.07   88.71   
2  151.0000(as on 30/11/2023)   2.44  10.06   4.59  0.88  ...   0.71   89.49   
3  151.0000(as on 30/11/2023)    2.4  10.01   4.69  0.88  ...   0.72   89.51   
4  276.1600(as on 28/02/2025)   0.62  -0.98  -1.61  0.96  ...    0.7   88.22   

  R2_5YRS R2_10YRS SD_3YRS SD_5YRS SD_10YRS SHP_3YRS SHP_5YRS SHP_10YRS  
0   88.06    90.02   16.12   16.66    18.82      0.2     0.99      0.34  
1   88.21    89.96   16.09   16.65    18.78     0.17     0.97      0.32  
2   73.71    59.24   18.84   23.07    20.99     0.75     1.66      0.64  
3   73.74    59.45   18.84   23.07    20.98     0.75     1.66      0.64  
4   82.52    85.55    14.3   13.04    12.47      0.5      1.0      0.29  

[5 rows x 21 columns]

In [17]:
# Cleaning Expense Ration Column
exp_ratio = []
exp = df_clean['ExpenseRatio']
exp = list(exp)
for i in exp:
    exp_ratio.append(float(i.split(' ')[0]))
df_clean['ExpenseRatio'] = exp_ratio

In [18]:
df_clean.head()

SchemeCode                                         SchemeName  \
0    100033.0  Aditya Birla Sun Life Equity Advantage Fund Gr...   
1    100034.0  Aditya Birla Sun Life Equity Advantage Fund Pa...   
2    100176.0  Quant Small Cap Fund Payout of Income Distribu...   
3    100177.0                        Quant Small Cap Fund Growth   
4    100218.0  JM Large Cap Fund Payout of Income Distributio...   

     SchemeCategory  Inception  ExpenseRatio               TurnoverRatio  \
0  Large & Mid- Cap 1995-02-24          1.95   40.0000(as on 28/02/2025)   
1  Large & Mid- Cap 1995-02-24          1.95   40.0000(as on 28/02/2025)   
2         Small-Cap 1996-10-31          1.64  151.0000(as on 30/11/2023)   
3         Small-Cap 1996-09-23          1.64  151.0000(as on 30/11/2023)   
4         Large-Cap 1995-01-04          2.25  276.1600(as on 28/02/2025)   

   ALP3Y  ALP5Y ALP10Y BET3Y  ... BET10Y R2_3YRS R2_5YRS R2_10YRS SD_3YRS  \
0   -4.5  -2.44  -2.13  1.04  ...   1.07   88.31   88.06    90.02   16.12   
1  -4.95  -2.76   -2.5  1.04  ...   1.07   88.71   88.21    89.96   16.09   
2   2.44  10.06   4.59  0.88  ...   0.71   89.49   73.71    59.24   18.84   
3    2.4  10.01   4.69  0.88  ...   0.72   89.51   73.74    59.45   18.84   
4   0.62  -0.98  -1.61  0.96  ...    0.7   88.22   82.52    85.55    14.3   

  SD_5YRS SD_10YRS SHP_3YRS SHP_5YRS SHP_10YRS  
0   16.66    18.82      0.2     0.99      0.34  
1   16.65    18.78     0.17     0.97      0.32  
2   23.07    20.99     0.75     1.66      0.64  
3   23.07    20.98     0.75     1.66      0.64  
4   13.04    12.47      0.5      1.0      0.29  

[5 rows x 21 columns]

In [19]:
# Cleaning Turnover Ration Column
turn_ratio = []
turn = df_clean['TurnoverRatio']
turn = list(turn)
for i in turn:
    turn_ratio.append(float(i.split('(')[0]))
df_clean['TurnoverRatio'] = turn_ratio
df_clean.head()

SchemeCode                                         SchemeName  \
0    100033.0  Aditya Birla Sun Life Equity Advantage Fund Gr...   
1    100034.0  Aditya Birla Sun Life Equity Advantage Fund Pa...   
2    100176.0  Quant Small Cap Fund Payout of Income Distribu...   
3    100177.0                        Quant Small Cap Fund Growth   
4    100218.0  JM Large Cap Fund Payout of Income Distributio...   

     SchemeCategory  Inception  ExpenseRatio  TurnoverRatio  ALP3Y  ALP5Y  \
0  Large & Mid- Cap 1995-02-24          1.95          40.00   -4.5  -2.44   
1  Large & Mid- Cap 1995-02-24          1.95          40.00  -4.95  -2.76   
2         Small-Cap 1996-10-31          1.64         151.00   2.44  10.06   
3         Small-Cap 1996-09-23          1.64         151.00    2.4  10.01   
4         Large-Cap 1995-01-04          2.25         276.16   0.62  -0.98   

  ALP10Y BET3Y  ... BET10Y R2_3YRS R2_5YRS R2_10YRS SD_3YRS SD_5YRS SD_10YRS  \
0  -2.13  1.04  ...   1.07   88.31   88.06    90.02   16.12   16.66    18.82   
1   -2.5  1.04  ...   1.07   88.71   88.21    89.96   16.09   16.65    18.78   
2   4.59  0.88  ...   0.71   89.49   73.71    59.24   18.84   23.07    20.99   
3   4.69  0.88  ...   0.72   89.51   73.74    59.45   18.84   23.07    20.98   
4  -1.61  0.96  ...    0.7   88.22   82.52    85.55    14.3   13.04    12.47   

  SHP_3YRS SHP_5YRS SHP_10YRS  
0      0.2     0.99      0.34  
1     0.17     0.97      0.32  
2     0.75     1.66      0.64  
3     0.75     1.66      0.64  
4      0.5      1.0      0.29  

[5 rows x 21 columns]

In [21]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 911 entries, 0 to 910
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   SchemeCode      911 non-null    float64       
 1   SchemeName      911 non-null    object        
 2   SchemeCategory  911 non-null    object        
 3   Inception       911 non-null    datetime64[ns]
 4   ExpenseRatio    911 non-null    float64       
 5   TurnoverRatio   911 non-null    float64       
 6   ALP3Y           911 non-null    object        
 7   ALP5Y           911 non-null    object        
 8   ALP10Y          911 non-null    object        
 9   BET3Y           911 non-null    object        
 10  BET5Y           911 non-null    object        
 11  BET10Y          911 non-null    object        
 12  R2_3YRS         911 non-null    object        
 13  R2_5YRS         911 non-null    object        
 14  R2_10YRS        911 non-null    object        
 15  SD_3YR

In [28]:
## choosing scheme which are 10 yrs old or more
df_clean = df_clean[df_clean['Inception'] <= '2015-03-31']
df_clean

SchemeCode                                         SchemeName  \
0      100033.0  Aditya Birla Sun Life Equity Advantage Fund Gr...   
1      100034.0  Aditya Birla Sun Life Equity Advantage Fund Pa...   
2      100176.0  Quant Small Cap Fund Payout of Income Distribu...   
3      100177.0                        Quant Small Cap Fund Growth   
4      100218.0  JM Large Cap Fund Payout of Income Distributio...   
..          ...                                                ...   
906    120475.0      Axis Strategic Bond Fund Direct Growth Option   
907    120477.0  Axis Conservative Hybrid Fund Direct Plan Annu...   
908    120478.0  Axis Conservative Hybrid Fund Direct Plan Half...   
909    120479.0  Axis Conservative Hybrid Fund Direct Plan Quar...   
910    120480.0  Axis Conservative Hybrid Fund Direct Plan Grow...   

              SchemeCategory  Inception  ExpenseRatio  TurnoverRatio  ALP3Y  \
0           Large & Mid- Cap 1995-02-24          1.95          40.00   -4.5   
1           Large & Mid- Cap 1995-02-24          1.95          40.00  -4.95   
2                  Small-Cap 1996-10-31          1.64         151.00   2.44   
3                  Small-Cap 1996-09-23          1.64         151.00    2.4   
4                  Large-Cap 1995-01-04          2.25         276.16   0.62   
..                       ...        ...           ...            ...    ...   
906          Medium Duration 2013-01-01          0.53         269.27   1.10   
907  Conservative Allocation 2013-11-09          0.90         126.91  -0.80   
908  Conservative Allocation 2013-05-07          0.90         126.91  -0.86   
909  Conservative Allocation 2013-01-21          0.90         126.91  -0.85   
910  Conservative Allocation 2013-01-01          0.90         126.91  -0.69   

     ALP5Y ALP10Y BET3Y  ... BET10Y R2_3YRS R2_5YRS R2_10YRS SD_3YRS SD_5YRS  \
0    -2.44  -2.13  1.04  ...   1.07   88.31   88.06    90.02   16.12   16.66   
1    -2.76   -2.5  1.04  ...   1.07   88.71   88.21    89.96   16.09   16.65   
2    10.06   4.59  0.88  ...   0.71   89.49   73.71    59.24   18.84   23.07   
3    10.01   4.69  0.88  ...   0.72   89.51   73.74    59.45   18.84   23.07   
4    -0.98  -1.61  0.96  ...    0.7   88.22   82.52    85.55    14.3   13.04   
..     ...    ...   ...  ...    ...     ...     ...      ...     ...     ...   
906   1.34   0.54  1.19  ...   1.26   83.91   83.11    62.70    1.37    1.55   
907   0.59  -1.38  1.17  ...   0.92   88.87   82.12    47.05    4.30    4.26   
908   1.23  -1.26  1.18  ...   0.98   89.23   77.74    50.62    4.32    4.47   
909   1.19  -1.19  1.17  ...   1.01   89.03   78.16    53.31    4.30    4.43   
910   0.68  -0.39  1.18  ...   0.98   88.91   82.19    57.04    4.32    4.27   

    SD_10YRS SHP_3YRS SHP_5YRS SHP_10YRS  
0      18.82      0.2     0.99      0.34  
1      18.78     0.17     0.97      0.32  
2      20.99     0.75     1.66      0.64  
3      20.98     0.75     1.66      0.64  
4      12.47      0.5      1.0      0.29  
..       ...      ...      ...       ...  
906     2.19     1.27     1.67      1.05  
907     4.85     0.44     1.09      0.24  
908     4.98     0.43     1.20      0.29  
909     4.98     0.43     1.20      0.32  
910     4.71     0.46     1.11      0.49  

[905 rows x 21 columns]

In [29]:
cols = df_clean.columns[6:]
for i in cols:
    print(df_clean[df_clean[i] == '–'][i])

471    –
815    –
816    –
Name: ALP3Y, dtype: object
471    –
815    –
816    –
Name: ALP5Y, dtype: object
465    –
471    –
472    –
528    –
529    –
532    –
535    –
539    –
632    –
716    –
720    –
734    –
770    –
773    –
775    –
815    –
816    –
852    –
876    –
900    –
Name: ALP10Y, dtype: object
471    –
815    –
816    –
Name: BET3Y, dtype: object
471    –
815    –
816    –
Name: BET5Y, dtype: object
465    –
471    –
472    –
528    –
529    –
532    –
535    –
539    –
632    –
716    –
720    –
734    –
770    –
773    –
775    –
815    –
816    –
852    –
876    –
900    –
Name: BET10Y, dtype: object
471    –
815    –
816    –
Name: R2_3YRS, dtype: object
471    –
815    –
816    –
Name: R2_5YRS, dtype: object
465    –
471    –
472    –
528    –
529    –
532    –
535    –
539    –
632    –
716    –
720    –
734    –
770    –
773    –
775    –
815    –
816    –
852    –
876    –
900    –
Name: R2_10YRS, dtype: object
471    –
815    –
816    –
Name: SD_3YRS, dtyp

In [36]:
indx = df_clean[(df_clean['ALP10Y']=='–') | (df_clean['BET10Y']=='–') | (df_clean['SD_10YRS']=='–') | (df_clean['R2_10YRS']=='–') | (df_clean['SHP_10YRS']=='–')].index
df_clean.drop(indx,inplace=True)

C:\Users\Chaitanya Keshari\AppData\Local\Temp\ipykernel_22964\555130686.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean.drop(indx,inplace=True)


In [37]:
## formatting the other performance and risk ratios
cols = df_clean.columns[6:]
for i in cols:
    df_clean[i] = df_clean[i].astype(float)
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 885 entries, 0 to 910
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   SchemeCode      885 non-null    float64       
 1   SchemeName      885 non-null    object        
 2   SchemeCategory  885 non-null    object        
 3   Inception       885 non-null    datetime64[ns]
 4   ExpenseRatio    885 non-null    float64       
 5   TurnoverRatio   885 non-null    float64       
 6   ALP3Y           885 non-null    float64       
 7   ALP5Y           885 non-null    float64       
 8   ALP10Y          885 non-null    float64       
 9   BET3Y           885 non-null    float64       
 10  BET5Y           885 non-null    float64       
 11  BET10Y          885 non-null    float64       
 12  R2_3YRS         885 non-null    float64       
 13  R2_5YRS         885 non-null    float64       
 14  R2_10YRS        885 non-null    float64       
 15  SD_3YR

C:\Users\Chaitanya Keshari\AppData\Local\Temp\ipykernel_22964\2801694085.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean[i] = df_clean[i].astype(float)


In [38]:
df_clean['SchemeCode'] = df_clean['SchemeCode'].astype(int)

C:\Users\Chaitanya Keshari\AppData\Local\Temp\ipykernel_22964\236867464.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean['SchemeCode'] = df_clean['SchemeCode'].astype(int)


In [39]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 885 entries, 0 to 910
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   SchemeCode      885 non-null    int32         
 1   SchemeName      885 non-null    object        
 2   SchemeCategory  885 non-null    object        
 3   Inception       885 non-null    datetime64[ns]
 4   ExpenseRatio    885 non-null    float64       
 5   TurnoverRatio   885 non-null    float64       
 6   ALP3Y           885 non-null    float64       
 7   ALP5Y           885 non-null    float64       
 8   ALP10Y          885 non-null    float64       
 9   BET3Y           885 non-null    float64       
 10  BET5Y           885 non-null    float64       
 11  BET10Y          885 non-null    float64       
 12  R2_3YRS         885 non-null    float64       
 13  R2_5YRS         885 non-null    float64       
 14  R2_10YRS        885 non-null    float64       
 15  SD_3YR

In [43]:
df_clean = df_clean.reset_index(drop=True)
df_clean

SchemeCode                                         SchemeName  \
0        100033  Aditya Birla Sun Life Equity Advantage Fund Gr...   
1        100034  Aditya Birla Sun Life Equity Advantage Fund Pa...   
2        100176  Quant Small Cap Fund Payout of Income Distribu...   
3        100177                        Quant Small Cap Fund Growth   
4        100218  JM Large Cap Fund Payout of Income Distributio...   
..          ...                                                ...   
880      120475      Axis Strategic Bond Fund Direct Growth Option   
881      120477  Axis Conservative Hybrid Fund Direct Plan Annu...   
882      120478  Axis Conservative Hybrid Fund Direct Plan Half...   
883      120479  Axis Conservative Hybrid Fund Direct Plan Quar...   
884      120480  Axis Conservative Hybrid Fund Direct Plan Grow...   

              SchemeCategory  Inception  ExpenseRatio  TurnoverRatio  ALP3Y  \
0           Large & Mid- Cap 1995-02-24          1.95          40.00  -4.50   
1           Large & Mid- Cap 1995-02-24          1.95          40.00  -4.95   
2                  Small-Cap 1996-10-31          1.64         151.00   2.44   
3                  Small-Cap 1996-09-23          1.64         151.00   2.40   
4                  Large-Cap 1995-01-04          2.25         276.16   0.62   
..                       ...        ...           ...            ...    ...   
880          Medium Duration 2013-01-01          0.53         269.27   1.10   
881  Conservative Allocation 2013-11-09          0.90         126.91  -0.80   
882  Conservative Allocation 2013-05-07          0.90         126.91  -0.86   
883  Conservative Allocation 2013-01-21          0.90         126.91  -0.85   
884  Conservative Allocation 2013-01-01          0.90         126.91  -0.69   

     ALP5Y  ALP10Y  BET3Y  ...  BET10Y  R2_3YRS  R2_5YRS  R2_10YRS  SD_3YRS  \
0    -2.44   -2.13   1.04  ...    1.07    88.31    88.06     90.02    16.12   
1    -2.76   -2.50   1.04  ...    1.07    88.71    88.21     89.96    16.09   
2    10.06    4.59   0.88  ...    0.71    89.49    73.71     59.24    18.84   
3    10.01    4.69   0.88  ...    0.72    89.51    73.74     59.45    18.84   
4    -0.98   -1.61   0.96  ...    0.70    88.22    82.52     85.55    14.30   
..     ...     ...    ...  ...     ...      ...      ...       ...      ...   
880   1.34    0.54   1.19  ...    1.26    83.91    83.11     62.70     1.37   
881   0.59   -1.38   1.17  ...    0.92    88.87    82.12     47.05     4.30   
882   1.23   -1.26   1.18  ...    0.98    89.23    77.74     50.62     4.32   
883   1.19   -1.19   1.17  ...    1.01    89.03    78.16     53.31     4.30   
884   0.68   -0.39   1.18  ...    0.98    88.91    82.19     57.04     4.32   

     SD_5YRS  SD_10YRS  SHP_3YRS  SHP_5YRS  SHP_10YRS  
0      16.66     18.82      0.20      0.99       0.34  
1      16.65     18.78      0.17      0.97       0.32  
2      23.07     20.99      0.75      1.66       0.64  
3      23.07     20.98      0.75      1.66       0.64  
4      13.04     12.47      0.50      1.00       0.29  
..       ...       ...       ...       ...        ...  
880     1.55      2.19      1.27      1.67       1.05  
881     4.26      4.85      0.44      1.09       0.24  
882     4.47      4.98      0.43      1.20       0.29  
883     4.43      4.98      0.43      1.20       0.32  
884     4.27      4.71      0.46      1.11       0.49  

[885 rows x 21 columns]

In [44]:
df_clean.to_csv('fodder2.csv',index = False)

In [46]:
df_clean.describe()

SchemeCode  ExpenseRatio  TurnoverRatio       ALP3Y       ALP5Y  \
count     885.000000    885.000000     885.000000  885.000000  885.000000   
mean   112573.568362      1.003503     145.120497    0.328339    0.672158   
std      7080.706298      0.596305     126.325737    2.363875    2.280061   
min    100033.000000      0.100000       0.000000   -6.530000   -5.090000   
25%    105670.000000      0.510000      43.150000   -1.010000   -0.570000   
50%    113179.000000      0.860000     112.280000   -0.250000    0.010000   
75%    119094.000000      1.540000     216.830000    1.080000    1.630000   
max    120480.000000      2.570000     633.980000    8.140000   10.100000   

           ALP10Y       BET3Y       BET5Y      BET10Y     R2_3YRS     R2_5YRS  \
count  885.000000  885.000000  885.000000  885.000000  885.000000  885.000000   
mean    -0.497322    1.699819    1.773548    1.012644   71.437130   65.716384   
std      1.404441    1.369319    1.552457    0.570717   26.922943   29.544310   
min     -4.540000   -5.730000   -1.690000   -6.030000    0.000000    0.000000   
25%     -1.310000    0.900000    0.920000    0.910000   61.890000   46.340000   
50%     -0.610000    1.000000    0.980000    0.990000   83.260000   77.830000   
75%      0.190000    2.290000    1.680000    1.160000   90.890000   89.280000   
max      4.690000    8.160000   11.120000    2.740000   98.780000   98.290000   

         R2_10YRS     SD_3YRS     SD_5YRS    SD_10YRS    SHP_3YRS    SHP_5YRS  \
count  885.000000  885.000000  885.000000  885.000000  885.000000  885.000000   
mean    59.027356    6.662554    7.147966    8.054282    0.274249    0.852011   
std     35.246928    6.549841    6.757085    7.609630    0.847802    0.644374   
min      0.000000    0.400000    0.510000    0.460000  -11.630000   -1.970000   
25%     20.680000    0.960000    1.270000    1.350000   -0.160000    0.480000   
50%     72.380000    3.000000    3.200000    4.020000    0.480000    1.000000   
75%     90.360000   13.850000   14.770000   16.660000    0.760000    1.310000   
max     98.070000   24.020000   23.070000   23.140000    6.270000    5.550000   

        SHP_10YRS  
count  885.000000  
mean     0.241763  
std      0.687583  
min     -3.350000  
25%     -0.040000  
50%      0.380000  
75%      0.540000  
max      2.470000

In [47]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 885 entries, 0 to 884
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   SchemeCode      885 non-null    int32         
 1   SchemeName      885 non-null    object        
 2   SchemeCategory  885 non-null    object        
 3   Inception       885 non-null    datetime64[ns]
 4   ExpenseRatio    885 non-null    float64       
 5   TurnoverRatio   885 non-null    float64       
 6   ALP3Y           885 non-null    float64       
 7   ALP5Y           885 non-null    float64       
 8   ALP10Y          885 non-null    float64       
 9   BET3Y           885 non-null    float64       
 10  BET5Y           885 non-null    float64       
 11  BET10Y          885 non-null    float64       
 12  R2_3YRS         885 non-null    float64       
 13  R2_5YRS         885 non-null    float64       
 14  R2_10YRS        885 non-null    float64       
 15  SD_3YR

## Debt-Oriented Funds
These are primarily suited for conservative investors seeking steady returns with lower risk.

1. Low Duration (130)

Invest in debt & money market instruments with a duration of 6 to 12 months.

Lower interest rate risk, good for short-term parking of funds.

2. Short Duration (77)

Duration: 1–3 years.

Balanced risk-return profile; better returns than liquid/low duration funds.

3. Money Market (64)

Invest in money market instruments (T-bills, CPs, CDs).

Tenure usually up to 1 year. Suitable for very short-term needs.

4. Ultra Short Duration (51)

Invest in instruments with 3–6 months duration.

Slightly higher returns than liquid funds with minimal volatility.

5. Medium Duration (39)

Duration: 3–4 years.

Moderate risk; impacted by interest rate changes.

6. Medium to Long Duration (39)

Duration: 4–7 years.

Higher interest rate risk; better during falling rate scenarios.

7. Corporate Bond (34)

Invest 80%+ in highest-rated corporate bonds.

Lower risk, stable returns.

8. Banking & PSU (41)

Invest in debt instruments of banks, PSUs, PFIs.

Good credit quality and safety.

9. 10 yr Government Bond (12)

Invest in government securities with 10-year maturity.

High interest rate sensitivity; suitable during rate cuts.

## Equity-Oriented Funds
Suitable for long-term investors seeking capital appreciation.

1. Large-Cap (82)

80%+ in top 100 companies by market cap.

Relatively stable, lower volatility.

2. Large & Mid-Cap (68)

Minimum 35% in both large- and mid-cap stocks.

Diversified exposure with growth potential.

3. Mid-Cap (57)

65%+ in mid-cap stocks (101–250th by market cap).

Higher risk and return potential.

4. Small-Cap (30)

65%+ in small-cap stocks (251+ ranked).

High risk, high return, volatile.

5. Multi-Cap (17)

25% each in large-, mid-, and small-cap.

Well-diversified across market caps.

6. Value (34)

Focus on undervalued stocks with turnaround potential.

Contrarian strategy; longer investment horizon needed.

7. Contra (12)

Invest against market trends.

Contrarian approach, requires patience and timing.